datasources/generic/generic.go

```diff
@@ -70,9 +70,56 @@ func (Client) Recognize(_ context.Context, _ []string) (timeline.Recognition, er
 	return timeline.Recognition{Confidence: 1}, nil
 }
 
+func (c *Client) processFile(ctx context.Context, path string, itemChan chan<- *timeline.Graph) (bool, error) {
+	fsys, err := archiver.FileSystem(ctx, path)
+	if err != nil {
+		return false, err
+	}
+
+	if _, isFileFS := fsys.(archiver.FileFS); !isFileFS {
+		return false, nil
+	}
```

Comment:
Since we're treating regular files differently, is the FileSystem needed? I usually use that only when I want to work with files, folders, and archives the same way regardless of what it is. That way I just have 1 code path for all those cases.

But if we're going to keep the refactor where regular files have a separate code flow, we can just skip this part, and stat the file to see what it is (already done below).

Reply:
So the thing here is that I wasn't sure how to detect the single file was not an archive, because it need to be walked then, other than having a map with file extensions that belong to archives and just check that.

Reply:
Oh right, that's a good point.

Then yeah, using a FileSystem is probably the way to go.